### PR TITLE
docs: api specs (integrations + internal)

### PR DESCRIPTION
**Pull request by mchalski**
Issues: MEN-793

includes some just-in-time design, e.g. regular token vs one-time token disctinction based on scope

Signed-off-by: mchalczynski <marcin.chalczynski@rndity.com>

@maciejmrowiec pls take a look - let's make sure we merge this today

@bboozzoo a quick look would be appreciated

@michaelatmender care to check style vs guildelines?

### CODE REVIEW

File: docs/integrations_api.yml [hunk 1]
```diff
@@ -0,0 +1,134 @@
+swagger: '2.0'
+info:
+  version: '0.1'
+  title: User administration and authentication
+  description: |
+    An API for user administration and user authentication handling. Intended for use by the web GUI.
+
+basePath: '/api/integrations/0.1/users'
```
Comment (mchalski) [inline docs/integrations_api.yml]: can't come up with a different human-friendly name. `useradmin`? `useradm` is too short. `user_management`?

File: docs/integrations_api.yml [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+          description: Internal server error.
+          schema:
+            $ref: '#/definitions/Error'
+
+definitions:
+  User:
+    description: New user descriptor.
+    type: object
+    properties:
+      name:
```
Comment (mchalski) [inline docs/integrations_api.yml]: ok, I guess Eystein wants editable emails :/ I'll have to change the `sub` claim in JWT to use the internal IDs.

Comment (mchalski) [inline docs/integrations_api.yml]: Also - screwed up user field descriptions. Updated.

Comment (maciejmrowiec) [inline docs/integrations_api.yml]: github uses usernames aka: maciejmrowiec, mensersofteare, mchalski in routes and during login, non editable
newrelic uses int id's in routes and email+pass for login, user name & email editable

Comment (mchalski) [inline docs/integrations_api.yml]: - in API routes we'll use (mongo) IDs
- in POST /users, the username is an email
- JWT also refers to ID
- login is based user + pass
- email and password are editable.

GET (future) will return both the ID and the current email.

Comment (maciejmrowiec) [inline docs/integrations_api.yml]: got it.


what about being more explicit call username => email maybe save some confusion for us

File: docs/integrations_api.yml (hunk 1 above)
Comment (maciejmrowiec) [inline docs/integrations_api.yml]: could be useradm just like a repo, burt don't have any particular solution, brought this up as I just noticed this `.../user/user/..`

Comment (mchalski) [inline docs/integrations_api.yml]: ok, let's go with useradm

File: docs/integrations_api.yml [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+      responses:
+        200:
+          description: |
+            Authentication successful - a new JWT is issued and returned.
+            The JWT is signed with the API's private key ('RS256' signing algorithm),
+            and contains the following standard claims:
+            * 'iss' - issuer
+            * 'exp' - expiry date
+            * 'sub' - unique, autogenerated user ID
+            * 'jti' - token's unique identifier (tracked for the purpose of revocation)
```
Comment (maciejmrowiec) [inline docs/integrations_api.yml]: do we plan to do token revocation? - it may make it harder to decentralise token verification in the future

Comment (mchalski) [inline docs/integrations_api.yml]: it's safe to remove `jti` for now. not sure if/when we decentralize, sth to discuss. 

also revocation is not that useful as with devices.

File: docs/internal_api.yml [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+  /auth/verify:
+    post:
+     summary: Check the validity of a token
+     description: |
+        Besides the basic validity check, checks the token expiration time and user-initiated token revocation.
+
+        Services which intend to use it should be correctly set up in the gateway's configuration.
+     parameters:
+       - name: token
+         in: body
```
Comment (bboozzoo) [inline docs/internal_api.yml]: headers

File: docs/internal_api.yml [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+paths:
+  /auth/verify:
+    post:
+     summary: Check the validity of a token
+     description: |
+        Besides the basic validity check, checks the token expiration time and user-initiated token revocation.
+
+        Services which intend to use it should be correctly set up in the gateway's configuration.
+     parameters:
+       - name: token
```
Comment (bboozzoo) [inline docs/internal_api.yml]: Authorization

Comment (mchalski) [inline docs/internal_api.yml]: done

File: docs/internal_api.yml (hunk 4 above)
Comment (mchalski) [inline docs/internal_api.yml]: done ('header')

**Comment by mchalski** [comment]
afaik all fixes are done. merge?

**Pull request merged**